ACG/AutoRouter.py

```diff
@@ -289,67 +289,76 @@ def draw_via(self,
-            # have what enclosure size
-            if enc_style == 'asymm':
-                # Determine whether the current route segment is on bottom or top
-                # Allocate the default enc params to the corresponding layer
-                if self.current_rect.get_highest_layer(layer=layer) == self.current_rect.lpp:
-                    default_enc = self.config['V' + layer + '_' + self.current_rect.layer]
-
-                    # Set the enclosure for the current route segment
-                    enc_large = default_enc['asymm_enclosure_large']
-                    enc_small = default_enc['asymm_enclosure_small']
-                    if self.current_dir == '+x' or self.current_dir == '-x':
-                        via.set_enclosure(enc_top=[enc_large, enc_large, enc_small, enc_small])
-                    else:
-                        via.set_enclosure(enc_top=[enc_small, enc_small, enc_large, enc_large])
-
-                    # Set the enclosure for the next route segment
-                    enc_large = default_enc['asymm_enclosure_large']
-                    enc_small = default_enc['asymm_enclosure_small']
-                    if direction == '+x' or direction == '-x':
-                        via.set_enclosure(enc_bot=[enc_large, enc_large, enc_small, enc_small])
-                    else:
-                        via.set_enclosure(enc_bot=[enc_small, enc_small, enc_large, enc_large])
-                else:
-                    default_enc = self.config['V' + self.current_rect.layer + '_' + layer]
-
-                    # Set the enclosure for the current route segment
-                    enc_large = default_enc['asymm_enclosure_large']
-                    enc_small = default_enc['asymm_enclosure_small']
-                    if self.current_dir == '+x' or self.current_dir == '-x':
-                        via.set_enclosure(enc_bot=[enc_large, enc_large, enc_small, enc_small])
-                    else:
-                        via.set_enclosure(enc_bot=[enc_small, enc_small, enc_large, enc_large])
-
-                    # Set the enclosure for the next route segment
-                    enc_large = default_enc['asymm_enclosure_large']
-                    enc_small = default_enc['asymm_enclosure_small']
-                    if direction == '+x' or direction == '-x':
-                        via.set_enclosure(enc_top=[enc_large, enc_large, enc_small, enc_small])
-                    else:
-                        via.set_enclosure(enc_top=[enc_small, enc_small, enc_large, enc_large])
-
-            # Set via parameters
-            if size is not None:
-                via.size = size
-            else:
-                via.size = self.config[via_id]['size']
-            if enc_bot is not None:
-                via.set_enclosure(enc_bot=enc_bot)
-            if enc_top is not None:
-                via.set_enclosure(enc_top=enc_top)
+        if self.current_dir[1] == 'x' and direction[1] == 'x':
```

Comment:
This is fine in cases where the default overlapping area between route segments is large enough, and the metals are wider than mid width. However, if we want to support mid-width routing, we need to be able to do the more complicated primitive via enclosures that the section below handles. Can we just integrate this into the section below and uncomment it instead?